src/pages/spawn/spawnHost/spawnHostModal/SetupScriptForm.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect } from "react";
+import { useLazyQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Checkbox from "@leafygreen-ui/checkbox";
+import { Input } from "antd";
+import { useLocation } from "react-router";
+import { GetTaskQuery, GetTaskQueryVariables } from "gql/generated/types";
+import { GET_TASK } from "gql/queries";
+import { parseQueryString, getString } from "utils";
+import { Action as SpawnHostModalAction } from "./useSpawnHostModalState";
+
+const { TextArea } = Input;
+
+export type setupScriptType = {
+  setUpScript?: string;
+  useProjectSetupScript?: boolean;
+  spawnHostsStartedByTask?: boolean;
+};
+
+interface SetupScriptFormProps {
+  onChange: React.Dispatch<SpawnHostModalAction>;
+  data: setupScriptType;
+}
+export const SetupScriptForm: React.FC<SetupScriptFormProps> = ({
+  onChange,
+  data,
+}) => {
+  const { setUpScript, useProjectSetupScript, spawnHostsStartedByTask } = data;
+
+  const { search } = useLocation();
+  const queryParams = parseQueryString(search);
+  const { taskId, distroId } = queryParams;
+  const [hasSetupScript, setHasSetupScript] = useState(false);
+
+  const [getTask, { data: taskData }] = useLazyQuery<
+    GetTaskQuery,
+    GetTaskQueryVariables
+  >(GET_TASK);
+
+  useEffect(() => {
+    getTask({ variables: { taskId: getString(taskId), execution: 0 } });
+    onChange({
+      type: "setProjectSetupScript",
+      taskId: getString(taskId),
+      useProjectSetupScript: true,
```

Comment:
should we hide the SetupScriptForm when taskId is unavailable? the mutation will error when taskId is nil and useProjectSetupScript is true: https://github.com/evergreen-ci/evergreen/blob/master/graphql/util.go#L890-L903

Reply:
Its still possible to provide a setup script when there isn't a task id provided. We hide the checkboxes that allow you to toggle a predefined setup script.

Reply:
When there is no task id useProjectSetupScript is false by default. and it wont even hit this condition in the resolver. https://github.com/evergreen-ci/evergreen/pull/4082/files#diff-a3ce50f20b746b196f29c6d414e685b0110c24b768bcf83756de8d842c98f6dcR369.
https://github.com/khelif96/spruce/blob/b75a89af621569d2541731ef80203053ff5428ee/src/pages/spawn/spawnHost/spawnHostModal/useSpawnHostModalState.ts#L44

It is not possible to toggle `useProjectSetupScript` to true  unless there is a task id provided, because of how the reducer is setup